### DISCUSSION TITLE
How to implement ViewPager-like navigation?

### DISCUSSION THREAD

**Discussion by alessandrofrancesconi** [AvaloniaUI/Avalonia#11604] [Q&A]
My data context is composed by N records, grouped in dates. Currently, my app shows only the records of the current day, in a vertical list. I would like the user to be able to horizontally scroll this list and change the date accordingly.
Basically, scrolling back shows you the records of "CurrentDate-1", scrolling forward for "CurrentDate+1".

I was thinking to implement this like a "infinite ViewPager" from Android: https://medium.com/@ashishpandey.professional/android-infinite-scrolling-viewpager2-195e3786ec3f

![image](https://github.com/AvaloniaUI/Avalonia/assets/1595223/56f39cb7-8841-4ddb-bb5d-884e07f509aa)

How to do something similar in Avalonia 11? Thanks for your help!

**Comment by workgroupengineering** [comment]
Usisng Carousel samples/ControlCatalog/Pages/CarouselPage.xaml
